pkg/aws/elb.go

```diff
@@ -20,42 +25,65 @@ func (c *Client) initELBClient() (*elb.ELB, error) {
 	if err != nil {
 		return nil, err
 	}
-	return elb.New(session), nil
-}
 
-func (c *Client) getELBDescriptions(eps []*pkg.Endpoint) ([]*elb.LoadBalancerDescription, error) {
-	client, err := c.initELBClient()
+	albs, err := c.getALBs(session)
 	if err != nil {
 		return nil, err
 	}
-	var names []*string
-	for _, ep := range eps {
-		elbName := extractELBName(ep.Hostname)
-		names = append(names, aws.String(elbName))
+
+	elbs, err := c.getELBs(session)
+	if err != nil {
+		return nil, err
 	}
-	params := &elb.DescribeLoadBalancersInput{
-		LoadBalancerNames: names,
-		// PageSize:          aws.Int64(1), use default 400 records
+
+	loadBalancers := append(albs, elbs...)
+	loadBalancersMap := map[string]string{} //map LB Dns to its canonical hosted zone id
+
+	for _, endpoint := range endpoints {
+		for _, loadBalancer := range loadBalancers {
+			if endpoint.Hostname == loadBalancer.DNSName {
+				loadBalancersMap[endpoint.Hostname] = loadBalancer.CanonZoneID
+			}
+		}
 	}
-	resp, err := client.DescribeLoadBalancers(params)
+
+	return loadBalancersMap, nil
+}
+
+func (c *Client) getELBs(session *session.Session) ([]*LoadBalancer, error) {
+	client := elb.New(session)
+
+	resp, err := client.DescribeLoadBalancers(nil)
 	if err != nil {
 		return nil, err
 	}
-	return resp.LoadBalancerDescriptions, nil
-}
 
-func getELBZoneID(ep *pkg.Endpoint, elbs []*elb.LoadBalancerDescription) *string {
-	for _, elb := range elbs {
-		if ep.Hostname == aws.StringValue(elb.DNSName) {
-			return elb.CanonicalHostedZoneNameID
-		}
+	loadBalancers := resp.LoadBalancerDescriptions
+	result := make([]*LoadBalancer, 0)
+	for _, loadbalancer := range loadBalancers {
+		result = append(result, &LoadBalancer{
+			DNSName:     aws.StringValue(loadbalancer.DNSName),
+			CanonZoneID: aws.StringValue(loadbalancer.CanonicalHostedZoneNameID),
```

Comment:
Any reason why we use an abbreviated name like `CanonZoneD` instead of `Canonical...`? Abbreviating normal words in code always looks strange to me.